### PR TITLE
Network 3

### LINKED ISSUE #49: Do not use the now-removed `Network` module
`HaskellNet` currently uses the function `connectTo`, from the `Network` module in the `network` library, to create its TCP connections.

In the `network` library, the `Network` module has long been deprecated. The docs say to use `Network.Socket` instead. Now the `Network` module has been removed, beginning with version 3.0.0.0 released in January 2019 (latest version is 3.0.0.1). It will soon become very difficult to compile `HaskellNet` in any up-to-date Haskell environment.

As an immediate stop-gap, please add a `network < 3.0.0.0` constraint to avoid build failures. Then please switch to using `Network.Socket` (or some other well-maintained networking library) to create the connections.

EDIT: Sorry, the `Network` module was not actually removed until 3.0.0.0, in January 2019. It's just that the haddocks stopped building in 2.7.0.0. But now it is gone altogether. Version numbers fixed.

### PR DESCRIPTION
**Pull request by phadej**
Resolves #49 and #53

### CODE REVIEW

File: HaskellNet.cabal [hunk 1]
```diff
@@ -54,12 +60,13 @@ Library
...
   Other-modules:
+    Network.Compat
     Text.Packrat.Pos
     Text.Packrat.Parse
 
   Build-Depends:
-    base >= 4.3 && < 4.13,
-    network >= 2 && < 3,
+    base >= 4.3 && < 4.14,
+    network >= 2.6.3.1 && < 3.2,
```
Comment (phadej) [inline HaskellNet.cabal]: https://hackage.haskell.org/package/network-2.6.3.1 is an most recent `network` which still supports GHC-7.0

It's released about three years ago, *at Sun Jul 31 03:48:32 UTC 2016*.

Comment (jtdaugherty) [reply HaskellNet.cabal]: Thanks for this info!

**Comment by phadej** [comment]
@jtdaugherty could you enable Travis CI (at https://travis-ci.org/jtdaugherty/HaskellNet) that would give nice green checks for good to go PRs

**Comment by phadej** [comment]
One point: the travis CI script **will** fail for old GHCs, as one of the checks is try to compile with versions of dependencies bundled with that particular GHC.

Due to `toStrict` (and `uncons` IIRC), we need somewhat recent `bytestring`. I'm personally ok with either way, either
- to shim `toStrict` (and `uncons`) which is simple (https://oleg.fi/gists/posts/2019-06-03-compat-packages.html#s:20)
- or simply require newer `bytestring` (and allow that in CI)

Either way, that would make another PR, but it's better done when CI is enabled.

---

After that, I don't see anything "urgent" to do. Then `master` would be in releasable state.

**Comment by jtdaugherty** [comment]
Travis should now be enabled.

**Comment by jtdaugherty** [comment]
I'm okay with just requiring a newer `bytestring`.

**Pull request merged**